tools/filesystem/file.go

```diff
@@ -87,6 +100,31 @@ func (r *PathReader) Open() (io.ReadSeekCloser, error) {
 
 // -------------------------------------------------------------------
 
+var _ FileReader = (*BufReader)(nil)
+
+type BufReader struct {
+	Bytes []byte
+}
+
+// Open implements the [filesystem.FileReader] interface.
+func (r *BufReader) Open() (io.ReadSeekCloser, error) {
+	return NewBytesReadSeekCloser(r.Bytes), nil
+}
+
+type BytesReadSeekCloser struct {
+	*bytes.Reader
+}
+
+func NewBytesReadSeekCloser(b []byte) *BytesReadSeekCloser {
+	return &BytesReadSeekCloser{bytes.NewReader(b)}
+}
+
+func (r *BytesReadSeekCloser) Close() error {
```

Comment:
Add a comment that this "implements" the `io.ReadSeekCloser` interface to be a little more clear why this method is here, eg:
```
Close implements the [io.ReadSeekCloser] interface.
```

_I'm still not sure whether it would be better reset/rewind the seeker, but let's keep it empty for now._